File: golang/1653. 使字符串平衡的最少删除次数.go
func minimumDeletions(s string) int {
    leftb, righta := 0, 0
    for _, c := range s {
        if c == 'a' {
            righta++
        }
    }
    res := righta
    for _, c := range s {
        if c == 'a' {
            righta--
        }else {
            leftb++
        }
        res = min(res, leftb + righta)
    }
    return res
}
func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: cpp/1653. 使字符串平衡的最少删除次数.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        int leftb=0,righta=0;
        for(auto ch:s){
            if(ch=='a'){
                righta++;
            }
        }
        int res=righta;
        for(int i=0;i<s.length();i++){
            if(s[i]=='a'){
                righta--;
            }else{
                leftb++;
            }
            res=min(res,leftb+righta);
        }
        return res;
    }
};